pluginutils.cpp

```diff
@@ -236,4 +237,33 @@ namespace pluginutils
 		return false;
 	}
 
+	int SetAmxString(AMX* amx, cell params, std::string_view str) 
+	{
+		cell* dest = reinterpret_cast<cell*>(amx->base + static_cast<int>(reinterpret_cast<AMX_HEADER*>(amx->base)->dat + params));
+		cell* start = dest;
+
+		std::copy(str.begin(), str.end(), dest);
+		return str.length();
+	}
+
+	// Author urShadow | Link: https://github.com/urShadow/Pawn.CMD/blob/master/src/Main.cpp#L684-L706
+	std::string GetAmxString(AMX* amx, cell params) 
```

Comment:
This code comes from a different author and is released under a more strict license (MIT) that requires attribution, which you didn't provide (as you didn't copy neither the copyright notice nor the license text from the original code). Did you ask urShadow for permission to use their code under a different license (zlib)? If not then I can't merge this code.

Reply:
Also, while this one wouldn't crash on invalid AMX address, but what if the user wants to print an error message if the address is invalid? How are they supposed to know if something went wrong? Sure, this function returns an empty string on failure, but an empty string may also be legitimately passed from the script side.

Reply:
> Also, while this one wouldn't crash on invalid AMX address, what if the user wants to print an error message if the address is invalid? How are they supposed to know if something went wrong? Sure, this function returns an empty string on failure, but an empty string may also be legitimately passed from the script side.

I agree. I suggest this solution (but it seems strange):
`return "(error)";`

Reply:
> This code comes from a different author and is released under a more strict license (MIT) that requires attribution, which you didn't provide (as you didn't copy neither the copyright notice nor the license text from the original code). Did you ask urShadow for permission to use their code under a different license (zlib)? If not then I can't merge this code.

No. I don't ask urShadow. I understand the license, but I would write similar code with unique_ptr. I thought it was a good idea to take its code (because it is already implemented through unique_ptr)

Reply:
>I agree. I suggest this solution (but it seems strange):
>`return "(error)";`

Which, again, can be absolutely legitimately passed from a script. You'll probably want to look at the other function definitions in `pluginutils.h` for a clue.